### ISSUE TITLE
Adding physics?

### ISSUE THREAD

**Issue by huytd** [huytd/agar.io-clone#28]
I'm thinking about adding physics engine for handling some animation (split animation,...) and for faster HitTest. 

But have no idea where should we apply the physics engine. On server side (more complex) or on client side (much easier? any risk?)

**Comment by Netifriik** [comment]
Animations and physics can be done client side. Anything that is not cosmetic like the splitting action itself should be server side but the client can handle the animations.

**Comment by huytd** [comment]
Yes, I think this much better for server and will not affect game performance.

**Comment by HansMusterWhatElse** [comment]
Physic engine should definitely not be applied on the client - that's actually one of the most important parts of the game and should be handled only by the server.

**Comment by MrJerB** [comment]
Probably obvious but hit-detection needs to remain on the server-side.

**Comment by abalabahaha** [comment]
I think this title makes more sense. Physics is more of the movement stuff, animations is more of the client-side transitions (cell growth, smoother movement, smoother eating, etc.)
